### DISCUSSION TITLE
LLVM buildtree size > 100GB?

### DISCUSSION THREAD

**Discussion by kwsp** [microsoft/vcpkg#37491] [Q&A]
Is this normal? I frequently run out of disk space when building LLVM with VCPKG.

`vcpkg/buildtrees/llvm/x64-windows-dbg/` size 99.9GB

Can I make VCPKG move to a difference location on a different drive? I couldn't find any documentation. I already have binary-cache on a different drive.

**Comment by kwsp** [comment]
![image](https://github.com/microsoft/vcpkg/assets/30088608/80c780db-6e95-4a3c-b42b-bd2be5052c95)

**Comment by StarGate-One** [comment]
Use `--x-buildtrees-root=<some-drive>/<some-path>` on the command line.

>->vcpkg help install
Synopsis: Installs a package
Examples:
  vcpkg install <port name> <port name>...
  vcpkg install zlib zlib:x64-windows curl boost
  vcpkg install --triplet x64-windows
See https://learn.microsoft.com/vcpkg/commands/install for more information.
Options:
  --allow-unsupported    Continues with a warning on unsupported ports, rather than failing
  --x-asset-sources=...  Asset caching sources. See 'vcpkg help assetcaching'
  --binarysource=...     Binary caching sources. See 'vcpkg help binarycaching'
  --x-buildtrees-root=...
                         Buildtrees directory (experimental)
  --clean-after-build    Cleans buildtrees, packages and downloads after building each package
  --clean-buildtrees-after-build
                         Cleans buildtrees after building each package
  --clean-downloads-after-build
                         Cleans downloads after building each package
  --clean-packages-after-build
                         Cleans packages after building each package
  --downloads-root=...   Downloads directory (default: %VCPKG_DOWNLOADS%)
  --dry-run              Does not actually build or install
  --editable             Disables source re-extraction and binary caching for libraries on the
                         command line (classic mode)
  --enforce-port-checks  Fails install if a port has detected problems or attempts to use a
                         deprecated feature
  --x-feature=...        Additional features from the top-level manifest to install (manifest mode)
  --head                 Installs the libraries on the command line using the latest upstream
                         sources (classic mode)
  --host-triplet=...     Host triplet. See 'vcpkg help triplet' (default:
                         %VCPKG_DEFAULT_HOST_TRIPLET%)
  --x-install-root=...   Installed directory (experimental)
  --keep-going           Continues installing packages on failure
  --x-no-default-features
                         Does not install the default features from the top-level manifest (manifest
                         mode)
  --no-downloads         Does not download new sources
  --no-print-usage       Does not print CMake usage information after install
  --only-binarycaching   Fails if cached binaries are not available
  --only-downloads       Makes best-effort attempt to download sources without building
  --overlay-ports=...    Directories of overlay ports (also: %VCPKG_OVERLAY_PORTS%)
  --overlay-triplets=... Directories of overlay triplets (also: %VCPKG_OVERLAY_TRIPLETS%)
  --x-packages-root=...  Packages directory (experimental)
  --recurse              Allows removal of packages as part of installation
  --triplet=...          Target triplet. See 'vcpkg help triplet' (default: %VCPKG_DEFAULT_TRIPLET%)
  --x-use-aria2          Uses aria2 to perform download tasks
  --vcpkg-root=...       The vcpkg root directory (default: %VCPKG_ROOT%)
  --x-write-nuget-packages-config=...
                         Writes a NuGet packages.config-formatted file for use with external binary
                         caching. See `vcpkg help binarycaching` for more information

**Reply by kwsp** [reply]
Thanks! Is there a way to specify this in an environmental variable? I'm mainly invoking vcpkg through the VS Code Cmake Tools extension, which makes using CMakePresets.json a breeze.

**Comment by StarGate-One** [comment]
Not that I know of.
